### PR TITLE
aws_compression_library_init()

### PR DESCRIPTION
**Pull request by graebm**
- add `aws_compression_library_init()`
- fix bug in `aws_huffman_decoder_reset()`
- fix encoder bug where `AWS_ERROR_SHORT_BUFFER` was returned even though we'd successfully encoded a full string

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,15 +7,17 @@ on:
       - '!master'
 
 env:
-  BUILDER_VERSION: v0.3.1
+  BUILDER_VERSION: v0.5.9
```
Comment (justinboswell) [inline .github/workflows/ci.yml]: v0.6.0

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -68,23 +70,40 @@ jobs:
...
+        docker run --env GITHUB_REF $DOCKER_IMAGE build -p ${{ env.PACKAGE_NAME }} --compiler=gcc-${{ matrix.version }}
 
   windows:
     runs-on: windows-latest
     steps:
     - name: Build ${{ env.PACKAGE_NAME }} + consumers
       run: |
-        python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_VERSION }}/builder', 'builder.pyz')"
-        python builder.pyz -p ${{ env.PACKAGE_NAME }} build default-downstream
+        python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
```
Comment (justinboswell) [inline .github/workflows/ci.yml]: '${{ env.BUILDER_HOST }}/releases/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}'

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -68,23 +70,40 @@ jobs:
...
+    steps:
+    - uses: ilammy/msvc-dev-cmd@v1
+      with:
+        toolset: 14.0
+        arch: ${{ matrix.arch }}
+        uwp: false
+        spectre: true
+    - name: Build ${{ env.PACKAGE_NAME }} + consumers
+      run: |
+        python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
```
Comment (justinboswell) [inline .github/workflows/ci.yml]: '${{ env.BUILDER_HOST }}/releases/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}'

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -68,23 +70,40 @@ jobs:
...
+        python builder.pyz build -p ${{ env.PACKAGE_NAME }} downstream
 
   osx:
     runs-on: macos-latest
     steps:
     - name: Build ${{ env.PACKAGE_NAME }} + consumers
       run: |
-        python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_VERSION }}/builder', 'builder')"
+        python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder')"
```
Comment (justinboswell) [inline .github/workflows/ci.yml]: '${{ env.BUILDER_HOST }}/releases/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}'

File: .github/workflows/ci.yml (hunk 1 above)
Comment (graebm) [reply .github/workflows/ci.yml]: I went to v0.0.6 but had to go back to v0.5.9. Seemed like some infinite recursion going on:

https://github.com/awslabs/aws-c-compression/runs/512841679
```
...
  File "<frozen importlib._bootstrap>", line 1007, in _handle_fromlist
RecursionError: maximum recursion depth exceeded in comparison
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
    Found project: aws-c-compression at /root/aws-c-compression
...
```

**Pull request merged**